src/vivarium_census_prl_synth_pop/tools/make_results.py

```diff
@@ -443,3 +445,41 @@ def generate_maps(
     #   }
 
     return maps
+
+
+def subset_results_by_state(processed_results_dir: str, state: str) -> None:
+    # Loads final results and subsets those files to a provide state excluding the Social Security Observer
+
+    from vivarium_census_prl_synth_pop.constants.metadata import US_STATE_ABBRV_MAP
+
+    abbrev_name_dict = {v: k for k, v in US_STATE_ABBRV_MAP.items()}
+    state_name = sanitize_location(abbrev_name_dict[state.upper()])
+    processed_results_dir = Path(processed_results_dir)
+    all_results = processed_results_dir / "usa"
+    state_dir = processed_results_dir / "states" / state_name
+    state_dir.mkdir(exist_ok=True, parents=True)
+    # copy files from final results to state directory for further processing
+    shutil.copytree(all_results, state_dir, dirs_exist_ok=True)
+
+    for observer in FINAL_OBSERVERS:
+        if observer == "social_security_observer":
+            shutil.rmtree(state_dir / observer)
+        logger.info(f"Processing data for {observer}...")
+        obs_dir = state_dir / observer
+        obs_files = sorted(
+            list(chain(*[obs_dir.glob(f"*.{ext}") for ext in SUPPORTED_EXTENSIONS]))
+        )
+        for file in obs_files:
+            df = read_datafile(file)
+            df["random_seed"] = file.name.split(".")[0].split("_")[-1]
+            df = formatter.format_columns(df)
+            if "state" in df.columns:
+                obs_data = df.loc[df["state"] == state]
+            elif "mailing_address_state" in df.columns:
+                obs_data = df.loc[df["mailing_address_state"] == state]
+            else:
+                file.unlink()
```

Comment:
Since we're deleting the SS observer directly above, we can remove this `file.unlink()` call. Have it raise an exception if neither of these columns exist.